File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/rerank.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import RerankRequest, RerankResponse

class RerankApi:
    """rerank API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def rerank(self, body: RerankRequest) -> RerankResponse:
        """Rerank documents"""
        return self._client.post(f"/ai/v3/rerank", json=body)

File: sdkwork-ai-sdk-flutter/lib/src/api/rerank.dart
import '../http/client.dart';
import '../models.dart';

class RerankApi {
  final HttpClient _client;
  
  RerankApi(this._client);

  /// Rerank documents
  Future<RerankResponse?> rerank(RerankRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/rerank'), body: body, contentType: 'application/json');
    return response is RerankResponse ? response : null;
  }
}
